src/map/ai/states/magic_state.cpp

```diff
@@ -182,7 +182,7 @@ bool CMagicState::CanCastSpell(CBattleEntity* PTarget)
     {
         return false;
     }
-    if (distance(m_PEntity->loc.p, PTarget->loc.p) > 40)
```

Comment:
28.5 is the range at which an in-progress mob spell should be canceled, and current code seems to work as expected.  But the maximum range at which a mob will begin to cast should be 26.8, and in current code because of the 40, ends up falling back on the 28.5 value also.  So I think ducky's right to change the 40 to a 26.8, but misleading in his comment that he tested to see if spells were interrupted past 27.